### DISCUSSION TITLE
node classification on elliptic bitcoin dataset

### DISCUSSION THREAD

**Discussion by SasikiranJ** [stellargraph/stellargraph#1832] [General]
Hi, I want to do semi supervised node classification on elliptic dataset. It has two labels 1(licit), 2(illicit) and unknown for unlabelled instances. how to represent class of  un-labelled instances to make model train on labeled set and predict on unknown label instances.

**Comment by adocherty** [comment]
Hi, thanks for your question.

A good starting point will be to look at the example [Node classification with Graph Convolutional Network](https://stellargraph.readthedocs.io/en/stable/demos/node-classification/gcn-node-classification.html).  This starts by splitting all the nodes randomly into train and test sets (for this example all nodes are labelled).

For your case, you would first want to filter the nodes into two sets: labelled and unlabelled using NumPy or Pandas. For example if you have numeric labels where `0` indicates the unlabelled class you can use Pandas to create these two sets as follows:
```
node_labels_labelled = node_subjects[node_labels != 0]
nodes_labels_unlabelled = node_subjects[node_labels == 0]
```

Note that you need the second set as the Pandas index contains the node IDs even though the labels will all be zero.

Then, the labelled nodes can be split into train and test sets in the same way as in the GCN example:
```
train_labels, test_labels = model_selection.train_test_split(node_labels_labelled, train_size=N)
```
where the number of nodes to be trained with `N` will depend on your dataset and GCN parameters.

Finally, after training and evaluating the test metrics the predictions can be obtained for the unlabelled nodes as follows:

```
unlabelled_flow = generator.flow(nodes_labels_unlabelled.index)
unlabelled_predictions = model.predict(unlabelled_flow)
```

Of course, as there are no known labels the accuracy of the predictions cannot be evaluated (hence it's good practice to split the labelled nodes into train and test)
